src/services/TechRecordsService.ts

```diff
@@ -215,6 +215,11 @@ class TechRecordsService {
   }
 
   private createAndArchiveTechRecord(techRecord: { vin: string, techRecord: ITechRecord[] }, msUserDetails: any) {
+    const {statusCode} = techRecord.techRecord[0];
+    delete techRecord.techRecord[0].statusCode;
+    if (statusCode === STATUS.ARCHIVED) {
```

Comment:
I think you should make sure that there are tech-records in the array that can be updated instead of just looking at the last one. I don't think the tech-records are ordered based on status.

Reply:
the techRecord that you see here is the payload. If the array would be empty it would fail in the updateTechRecords function and not even reach the service.
if (!techRec || !techRec.length) {
    return Promise.resolve(new HTTPResponse(400, "Body is not a valid TechRecord"));
  }